optimum_benchmark/backends/pytorch.py

```diff
@@ -268,128 +340,75 @@ def prepare_for_profiling(
         LOGGER.info("\t+ Wrapping model with FXProfilingWrapper")
         self.pretrained_model = FXProfilingWrapper(self.pretrained_model)
 
-    def forward(self, input: Dict[str, Tensor], **kwargs) -> ModelOutput:
+    def forward(self, input: Dict[str, Tensor], **kwargs) -> "ModelOutput":
         with torch.autocast(
+            enabled=self.config.amp_autocast,
             device_type=self.device.type,
-            dtype=self.amp_dtype,
-            enabled=self.amp_autocast,
+            dtype=self.config.amp_dtype,
         ):
             output = self.pretrained_model(**input, **kwargs)
 
         return output
 
-    def generate(self, input: Dict[str, Tensor], **kwargs) -> ModelOutput:
+    def generate(self, input: Dict[str, Tensor], **kwargs) -> "ModelOutput":
         with torch.autocast(
+            enabled=self.config.amp_autocast,
             device_type=self.device.type,
-            dtype=self.amp_dtype,
-            enabled=self.amp_autocast,
+            dtype=self.config.amp_dtype,
         ):
             output = self.pretrained_model.generate(**input, **kwargs)
 
         return output
 
-    def train(self) -> None:
-        raise Exception("For PyTorch backend training, please call backend.run_pytorch_training.")
-
-    def run_pytorch_training(self, training_config, training_arguments, training_dataset, training_data_collator):
-        LOGGER.info("Running training benchmark")
-
-        # Converting from DictConfig to Dict is required to avoid a warning with DDP:
-        # `[W CudaIPCTypes.cpp:15] Producer process has been terminated before all shared CUDA tensors released. See Note [Sharing CUDA tensors]`
-        training_arguments_dict = OmegaConf.to_container(training_arguments, resolve=True)
-        
-        if training_config.use_ddp:
-            # TODO: support multi-node training. Hydra is probably not the good infra for that though.
-            if training_config.ddp_config.max_nodes != 1:
-                raise ValueError("PyTorch DDP training benchmark currently supports only training on a single node.")
-                
-            launch_config = LaunchConfig(**training_config.ddp_config)
-            LOGGER.info(f"PyTorch DDP launch config: {launch_config}")
-            
-            # TODO: The backend instance can not be passed here (cannot pickle 'weakref' object) so the nn.Module is passed directly.
-            # It is not clear who is using weakref though.
+    @record
```

Comment:
What is this doing?

Reply:
batter logging for pytorch multiprocessing

Reply:
Do you have an example?